src/sdb.c

```diff
@@ -449,7 +449,7 @@ static int sdb_set_internal (Sdb* s, const char *key, char *val, int owned, ut32
 			if (cas && kv->cas != cas) {
 				return 0;
 			}
-			if (vlen == kv->value_len && !strcmp (kv->value, val)) {
+			if (vlen == kv->value_len && strcmp (kv->value, val) != 0) {
```

Comment:
Okay, would you like me to revert this change and use this pull request to fix the failing tests? In addition, I think fixing the tests to exit 1 would be a good idea to detect this earlier.